workflows/patron_hold_request.md

```diff
@@ -10,21 +10,27 @@ Due to both this approach, and the circumstances under which the LSP was built,
 
-4. The difference between a request being sent through the SCC and SCSB UI is that in the former case the hold request has a ‘pickup location’ and in the latter it has a ‘delivery location’. This becomes significant in step 5.
+4. The difference between a request being sent through the RC and SCSB UI is that in the former case the hold request has a ‘pickup location’ (Sierra location id) and in the latter it has a ‘delivery location’ (ReCAP customer code). This becomes significant in step 5.
 
 5. The HoldRequestService stores details of the Hold request (i.e., the patron, item, whether EDD or delivered, and if the latter a delivery location) and enters an event into the HoldRequest Kinesis stream.
 
-6. That event is picked up by the HoldRequestConsumer, whose job it is to hit the SCSB API to pass on the Hold Request, only if the request has a ‘pickup location’ or was an EDD request. I.e., it only passes the request to SCSB if the request originated in SCC — if it originated in SCSB, their system already knows about it so we do not need to tell them about it through the API.
+6. That event is picked up by the HoldRequestConsumer, whose job it is to push the request to one of two places depending on if the item is on- or off-site.
 
-7. In order to hit the SCSB API, the HoldRequestConsumer makes a request to the PatronService to change the patron id for the barcode
+7. If the request is for an on-site item, the HoldRequestConsumer will forward the request to the OnSiteHoldRequestService. This includes phys and edd requests. Note that requests for on-site items implicitly originated in RC since SCSB has no knowledge of our on-site items.
 
-8. Finally, if that request was either a successful EDD request or was unsuccessful (two conditions where there is nothing more to process downstream), HoldRequestConsumer pushes the result to the HoldRequestResult event stream.
+   * The OnSiteHoldRequestService handles phys and edd requests for on-site items. When the incoming hold is a physical request, it uses the Sierra API to create the hold on the patron's account, and returns the response to the caller. If the incoming request is EDD, it generates a LibAnswers email with information about the request.
+
+8. If the request is for an off-site item, the HoldRequestConsumer will forward the request to SCSB. Critiera for pushing the hold request to SCSB:
+   - The request is for an off-site item (i.e. an item whose location id matches `/^rc/`)
+   - The hold originated in RC (i.e. has a `pickupLocation` or is EDD). If the hold originated in SCSB, their system already knows about it so we do not need to tell them about it through the API
+
+9. Finally, if that request was either a successful EDD request, a successful on-site request, or was /unsuccessful/, (three conditions where there is nothing more to process downstream) HoldRequestConsumer pushes the result to the HoldRequestResult event stream.
```

Comment:
can you specify what the other scenarios are that would not end up getting pushed downstream?